### PR TITLE
Show Answer Details in Assessment Statistics Page (Part 1: Last Graded Answer (MCQ, MRQ, Text Response, File Upload))

### PR DESCRIPTION
**Pull request by bivanalhar**
## Background

This is the continuation of the previous PR https://github.com/Coursemology/coursemology2/pull/7077 and https://github.com/Coursemology/coursemology2/pull/7084 that adds feature allowing teacher to see the details of the answer when clicking on the grade box in the Assessment Statistics Page

This PR is related with the issue https://github.com/Coursemology/coursemology2/issues/4567

## Feature Details

The display of the answer is as follows:
Screenshot 2024-08-01 at 5 09 38 PM

How to navigate:
- In either Marks per Question Table or Attempt Count Table (available inside the Assessment Statistics Page), you might click on any grade/attempt-count box.
- The answer details pop-up box will be shown immediately after.

Some consideration in the feature implementation:
- Question is included in the Details as a reference for the teacher to make them easier to look into the question details while looking at the Answer.
- Since our focus here is the details inside the answer, we made Question to be collapsible, with not expanded as the default.
- We define the new component instead of redirecting the user to the Submission Page so that user can stay on the statistics page while examining the Answer

## TO-DO Lists

- [x] Define component for Forum Post Response and Programming Answer (higher priority) (https://github.com/Coursemology/coursemology2/pull/7131)
- [x] Get all past answers (other attempts made by students other than the last graded one) and display the most recent 10 answers into the Answer Details component (https://github.com/Coursemology/coursemology2/pull/7135)
- [x] Define new page consisting of all the past answers list (possibly creating new API for this as well) (https://github.com/Coursemology/coursemology2/pull/7135)

## Future Improvement

- Define component for Voice Response and Scribing (at this point, not urgent to do)

## Additional Notes

The codecov here noted some of the changes in BE not covered in the tests in RSpec. However, this will be later covered in the PR https://github.com/Coursemology/coursemology2/pull/7135

### CODE REVIEW

File: client/app/bundles/course/assessment/pages/AssessmentStatistics/AnswerDetails/AnswerDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import FileUploadDetails from './FileUploadDetails';
+import MultipleChoiceDetails from './MultipleChoiceDetails';
+import MultipleResponseDetails from './MultipleResponseDetails';
+import TextResponseDetails from './TextResponseDetails';
+
+const translations = defineMessages({
+  rendererNotImplemented: {
+    id: 'course.assessment.submission.Answer.rendererNotImplemented',
+    defaultMessage:
+      'The display for this question type has not been implemented yet.',
```
Comment (cysjonathan) [inline client/app/bundles/course/assessment/pages/AssessmentStatistics/AnswerDetails/AnswerDetails.tsx:18]: What's the difference between
`course.assessment.submission.Submission.Answer.rendererNotImplemented`
and
`course.assessment.submission.Submission.SubmissionAnswer.rendererNotImplemented

File: client/app/bundles/course/assessment/pages/AssessmentStatistics/AnswerDetails/TextResponseDetails.tsx [hunk 2]
```diff
@@ -14,7 +14,7 @@ const TextResponseDetails = (
         dangerouslySetInnerHTML={{ __html: answer.fields.answer_text }}
         variant="body2"
       />
-      {question.allowAttachment && (
+      {question.maxAttachments > 0 && (
```
Comment (cysjonathan) [inline client/app/bundles/course/assessment/pages/AssessmentStatistics/AnswerDetails/TextResponseDetails.tsx:17]: This should have been squashed with 2nd previous commit

**Review by cysjonathan — CHANGES_REQUESTED** [review]

File: client/app/bundles/course/assessment/pages/AssessmentStatistics/AnswerDetails/AnswerDetails.tsx (hunk 1 above)
Comment (bivanalhar) [reply client/app/bundles/course/assessment/pages/AssessmentStatistics/AnswerDetails/AnswerDetails.tsx:18]: actually the inconsistencies already occur quite long ago, starting from when SubmissionAnswer was being refactored to Answer (that time, the focus was to make the implementation of Answer to be more concise and more expandable towards the future attributes, and hence the translation part is overlooked).

Now after the thorough observation, it's only natural to change the translation all from SubmissionAnswer to Answer, to comply with the new naming of that component, which is Answer

File: client/app/bundles/course/assessment/pages/AssessmentStatistics/AnswerDetails/TextResponseDetails.tsx (hunk 2 above)
Comment (bivanalhar) [reply client/app/bundles/course/assessment/pages/AssessmentStatistics/AnswerDetails/TextResponseDetails.tsx:17]: done

**Pull request merged**